apps/meteor/tests/unit/server/startup/initialData.tests.ts

```diff
@@ -0,0 +1,235 @@
+	'../../app/lib/server/functions/checkUsernameAvailability': {
+		checkUsernameAvailability,
+	},
+	'../../app/settings/server': {
+		settings: { get: settingsGet },
+	},
+	'../../lib/emailValidator': {
+		validateEmail,
+	},
+	'../lib/roles/addUserRoles': {
+		addUserRolesAsync,
+	},
+	'@rocket.chat/models': models,
+});
+
+describe('insertAdminUserFromEnv', () => {
+	beforeEach(() => {
+		getUsersInRole.reset();
+		checkUsernameAvailability.reset();
+		validateEmail.reset();
+		addUserRolesAsync.reset();
+		models.Users.create.reset();
+		models.Users.findOneByEmailAddress.reset();
+		setPasswordAsync.reset();
+		settingsGet.reset();
+		process.env.ADMIN_PASS = 'pass';
+	});
+
+	it('should do nothing if process.env.ADMIN_PASS is empty', async () => {
+		process.env.ADMIN_PASS = '';
+		const result = await insertAdminUserFromEnv();
+		expect(getUsersInRole.called).to.be.false;
+		expect(result).to.be.undefined;
+	});
+	it('should do nothing if theres already an admin user', async () => {
+		getUsersInRole.returns({ count: () => 1 });
+
+		const result = await insertAdminUserFromEnv();
+		expect(getUsersInRole.called).to.be.true;
+		expect(validateEmail.called).to.be.false;
+		expect(result).to.be.undefined;
+	});
+	it('should try to validate an email when process.env.ADMIN_EMAIL is set', async () => {
+		process.env.ADMIN_EMAIL = 'email';
+		getUsersInRole.returns({ count: () => 0 });
+		validateEmail.returns(false);
+		models.Users.create.returns({ insertedId: 'newuserid' });
+
+		const result = await insertAdminUserFromEnv();
+		expect(getUsersInRole.called).to.be.true;
+		expect(validateEmail.called).to.be.true;
+		expect(validateEmail.calledWith('email')).to.be.true;
+		expect(models.Users.create.called).to.be.true;
+		expect(setPasswordAsync.called).to.be.true;
+		expect(result).to.be.undefined;
+	});
+	it('should override the admins name when process.env.ADMIN_NAME is set', async () => {
+		process.env.ADMIN_EMAIL = 'email';
+		process.env.ADMIN_NAME = 'name';
+		getUsersInRole.returns({ count: () => 0 });
+		validateEmail.returns(true);
+		validateEmail.returns(false);
+		models.Users.create.returns({ insertedId: 'newuserid' });
+
+		await insertAdminUserFromEnv();
+
+		expect(
+			models.Users.create.calledWith(
+				sinon.match({
+					name: 'name',
+					username: 'admin',
+					status: 'offline',
+					statusDefault: 'online',
+					utcOffset: 0,
+					active: true,
+					type: 'user',
+				}),
+			),
+		).to.be.true;
+	});
+	it('should ignore the admin email when another user already has it set', async () => {
+		process.env.ADMIN_EMAIL = 'email';
+		getUsersInRole.returns({ count: () => 0 });
+		validateEmail.returns(true);
+		models.Users.create.returns({ insertedId: 'newuserid' });
+		models.Users.findOneByEmailAddress.returns({ _id: 'someuser' });
+
+		await insertAdminUserFromEnv();
+
+		expect(models.Users.findOneByEmailAddress.getCall(0).firstArg).to.not.to.have.property('email', 'email');
```

Comment:
Wouldn't it be better to check if the user is being created with the proper email instead? Like we are doing here: https://github.com/RocketChat/Rocket.Chat/pull/32066/files#diff-bae5a355e8a182e2302bf4cb73656f039b36c9dd87884218b724e3529e1a5e9eR177

Reply:
ops, wrong stub